### PR TITLE
Add warning message to `session save` when transcript isn't saved.

### PR DESCRIPTION
**Pull request by zhyty**
Somewhat recently, we made the change to hide the behavior to save LLDB session history to the transcript buffer behind the flag `interpreter.save-transcript`. By default, `interpreter.save-transcript` is false. See #90703 for context.

I'm making a small update here to our `session save` messaging and some help docs to clarify for users that aren't aware of this change. Maybe `interpreter.save-transcript` could be true by default as well. Any feedback welcome.

# Tests
```
bin/lldb-dotest -p TestSessionSave
```

### CODE REVIEW

File: lldb/source/Interpreter/CommandInterpreter.cpp [hunk 1]
```diff
@@ -3306,6 +3306,8 @@ bool CommandInterpreter::SaveTranscript(
   result.SetStatus(eReturnStatusSuccessFinishNoResult);
   result.AppendMessageWithFormat("Session's transcripts saved to %s\n",
                                  output_file->c_str());
+  if (!GetSaveTranscript())
+    result.AppendError("Note: the setting interpreter.save-transcript is set to false, so the transcript might not have been recorded.");
```
Comment (jeffreytan81) [inline lldb/source/Interpreter/CommandInterpreter.cpp:3310]: I am not sure why we are using `might not` here. Since `GetSaveTranscript()` is false, isn't the transcript definitely won't be saved? Maybe rewording to: 

```
Note: the interpreter.save-transcript setting is disabled (false), so the transcript will not be saved.
```

**Review by jeffreytan81 — APPROVED** [review]

File: lldb/source/Interpreter/CommandInterpreter.cpp (hunk 1 above)
Comment (medismailben) [inline lldb/source/Interpreter/CommandInterpreter.cpp:3310]: We should do this at the beginning of the function, before doing the work and return early if `save-transcript` is disabled.

**Review by medismailben — CHANGES_REQUESTED** [review]

**Comment by medismailben** [comment]
> Maybe `interpreter.save-transcript` could be true by default as well. Any feedback welcome.

Although that could be useful but I'm concerned of the performance aspect of it. Not everyone cares to save transcripts so I'm not sure it should be on by default. I like this patch because it explains to the user what they need to do to save the transaction (enable the settings), may be that's good enough.

File: lldb/source/Interpreter/CommandInterpreter.cpp (hunk 1 above)
Comment (zhyty) [reply lldb/source/Interpreter/CommandInterpreter.cpp:3310]: Here's an example:
```
(lldb) settings set interpreter.save-transcript true
(lldb) p 1
(int) 1
(lldb) hello
error: 'hello' is not a valid command.
(lldb) settings set interpreter.save-transcript false
(lldb) p 2
(int) 2
(lldb) session save
Session's transcripts saved to /var/folders/dh/nv6c_38j4tq3crp6r6f69sdc0000gn/T/lldb/lldb_session_2024-09-20_17-46-03.936239000.log
error: Note: the setting interpreter.save-transcript is set to false, so the transcript might not have been recorded.
```

In this case, the text in between the two `interpreter.save-transcript` settings will be saved, including the actual command to disable it. So here's what's saved:
```
(lldb) settings set interpreter.save-transcript true
(lldb) p 1
(int) 1
(lldb) hello
[0;1;31merror: [0m'hello' is not a valid command.
(lldb) settings set interpreter.save-transcript false
```

This is why I worded it like this. `session save` *will* save whatever's found in the transcript buffer, but what was actually saved in the transcript buffer depends if users did something like this.

Comment (zhyty) [reply lldb/source/Interpreter/CommandInterpreter.cpp:3310]: Sounds good to me. I'll make the change.

Comment (zhyty) [reply lldb/source/Interpreter/CommandInterpreter.cpp:3310]: See my [comment above](https://github.com/llvm/llvm-project/pull/109020#discussion_r1769375600).

I hate to be so pedantic about it, but I would be worried if users used this flow of 
```
(lldb) settings set interpreter.save-transcript true
... do various things they want to save ...
(lldb) settings set interpreter.save-transcript false
... do various things they don't want to save ...
(lldb) session save 
```
This wouldn't work anymore if we early returned.

To your point though, I guess they could always re-toggle it before that last session save, and maybe it would be better for the average user.

Anyway, that's why I decided to make this commit so conservative.

Comment (jeffreytan81) [reply lldb/source/Interpreter/CommandInterpreter.cpp]: Hmm, one would argue that when `interpreter.save-transcript` is set to false, it should clear any internal buffer because the name `interpreter.save-transcript = false` implies it should not save transcript. 

For the current behavior, `interpreter.save-transcript` should be renamed to "stop-record-commands" or something like this. 

Not say it is important or I care though.

Comment (zhyty) [reply lldb/source/Interpreter/CommandInterpreter.cpp]: @medismailben friendly ping on any followup comments. If not, I'll go ahead and merge.
